src/conv-core/conv-conds.C

```diff
@@ -10,21 +10,32 @@ CpvDeclare(double, idleBeginWalltime); // used for determining the conditon for
 #endif
 
 /**
- * Structure to hold the requisites for a callback
+ * Structure to hold the requisites for a conditional callback
  */
-typedef struct _ccd_callback {
+typedef struct _ccd_cond_callback {
+  CcdCondFn fn;
+  void *arg;
+  int pe;			/* the pe that sets the callback */
+} ccd_cond_callback;
+
+
+
+/**
+ * Structure to hold the requisites for a periodic callback
+ */
+typedef struct _ccd_periodic_callback {
   CcdVoidFn fn;
   void *arg;
   int pe;			/* the pe that sets the callback */
-} ccd_callback;
+} ccd_periodic_callback;
 
 
 
 /**
  * An element (a single callback) in a list of callbacks
  */
 typedef struct _ccd_cblist_elem {
```

Comment:
Maybe the typename of this struct (and for `ccd_heap_elem`) should now reflect that they only accept callback of their respective types? Not a big deal, though.